src/parser.rs

```diff
@@ -0,0 +1,196 @@
+use crate::expr::{Expr, Unary};
+use crate::parse_error::ParseError;
+use crate::token::token::Token;
+use crate::token::token_type::TokenType;
+
+#[derive(Default)]
+pub struct Parser {
+    pub tokens: Vec<Token>,
+    pub current: usize,
+}
+
+impl Parser {
+    fn expression(self) -> Expr {
+        return self.equality();
+    }
+
+    fn synchronize(mut self) {
+        self.advance();
+
+        while !self.is_at_end() {
+            if self.previous().unwrap().token_type == TokenType::SEMICOLON {
+                return;
+            }
+
+            let ignored_tokens = vec![
```

Comment:
Storing this in a vec is a little strange, because you're likely allocating memory here that you needn't. I think you'd be better off using pattern matching (so `if matches!(thing, TokenType::CLASS | TokenType::FUN...`), and maybe putting that into a method on TokenType or something.